x-pack/plugins/endpoint/public/embeddables/resolver/lib/tree_sequencers.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// TODO: type root and children
+export function* depthFirstPreorder({ root, children }) {
+  const nodesToVisit = [root];
+  while (nodesToVisit.length !== 0) {
+    const currentNode = nodesToVisit.shift();
+    nodesToVisit.unshift(...(children(currentNode) || []));
+    yield currentNode;
+  }
+}
+
+export function* levelOrder({ root, children }) {
```

Comment:
These look good, but I think having "nodes" in the name of the function would make it easier to understand (like `nodesByLevelOrder` or `levelOrderNodes`)